### PR TITLE
Unify and crossref docstrings for `GF` and `finite_field`

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/oscar-system/Oscar.jl/issues/3669.

### CODE REVIEW

File: src/flint/fq_default.jl [hunk 1]
```diff
@@ -842,16 +842,19 @@ end
...
+    finite_field(p::IntegerUnion, d::Int, s::VarName = :o; cached::Bool = true, check::Bool = true)
+    finite_field(q::IntegerUnion, s::VarName = :o; cached::Bool = true, check::Bool = true)
+    finite_field(f::FqPolyRingElem, s::VarName = :o; cached::Bool = true, check::Bool = true)
 
-Return a tuple $K, a$ consisting of a finite field $K$ of order $q = p^d$ and
-algebra generator $x$. The string $s$ is used to designate how the finite field
-generator will be printed.
+Return a tuple $(K, x)$ of a finite field $K$ of order $q = p^d$, where $p$ is a prime,
+and an algebra generator $x$ of $K$.
+The string $s$ is used to designate how the finite field generator will be printed.
```
Comment (fingolfin) [inline src/flint/fq_default.jl:851]: "finite field generator" = "algebra generator" = "x" ?

Potentially confusing?

**Review by fingolfin — APPROVED** [review]
SingularCI tests fail due to https://github.com/oscar-system/Singular.jl/issues/811

File: src/flint/fq_default.jl [hunk 2]
```diff
@@ -893,17 +896,19 @@ function finite_field(f::FqPolyRingElem, s::VarName = :o; cached::Bool = true, c
...
-Return a finite field $K$ of order $q = p^d$. The string $s$ is
-used to designate how the finite field generator will be printed.
+Return a finite field $K$ of order $q = p^d$, where $p$ is a prime.
+The identifier $s$ is used to designate how the finite field generator will be printed.
 
 If a polynomial $f \in k[X]$ over a finite field $k$ is specified,
 the finite field $K = k[X]/(f)$ will be constructed as a finite
 field with base field $k$.
 
+See also [`finite_field`](@ref) which additionally returns a finite field generator generator of $K$.
```
Comment (fingolfin) [inline src/flint/fq_default.jl:910]:
```suggestion
See also [`finite_field`](@ref) which additionally returns a finite field generator of $K$.
```

File: src/flint/fq_default.jl [hunk 3]
```diff
@@ -842,16 +842,19 @@ end
...
-    finite_field(f::FqPolyElem, s::VarName; cached = true, check = true)
+    finite_field(p::IntegerUnion, d::Int, s::VarName = :o; cached::Bool = true, check::Bool = true)
+    finite_field(q::IntegerUnion, s::VarName = :o; cached::Bool = true, check::Bool = true)
+    finite_field(f::FqPolyRingElem, s::VarName = :o; cached::Bool = true, check::Bool = true)
 
-Return a tuple $K, a$ consisting of a finite field $K$ of order $q = p^d$ and
-algebra generator $x$. The string $s$ is used to designate how the finite field
-generator will be printed.
+Return a tuple $(K, x)$ of a finite field $K$ of order $q = p^d$, where $p$ is a prime,
+and a finite field generator $x$ of $K$.
```
[the remark this answers is not in the document]
Comment (lgoettgens) [reply src/flint/fq_default.jl:850]: I don't really see a great way how to put all of this into this docstring without making it overly long. Thus I think I would keep this to the future. (Unless you come up with something in the next days or so)

[the remark this answers is not in the document]
Comment (fingolfin) [reply src/flint/fq_default.jl:850]: Perhaps this could just say it as I suggested, and then we make sure that `base_field(K)` has a helpful docstring.

**Comment by thofma** [comment]
In some other place we rephrased this "algebra generator", see the `?gen`:
```
    gen(L::FqField)

  Return a K-algebra generator a of the finite field L, where K is the base field of L. The element a satisfies
  defining_polyomial(a) == 0.

  Note that this is in general not a multiplicative generator and can be zero, if L/K is an extension of degree
  one.
```

File: src/flint/fq_default.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (fingolfin) [reply src/flint/fq_default.jl:850]: Following Tommy's comment, how about
```suggestion
and a generator $x$ of $K$ (see [`gen`](@ref) for a definition).
```

[the remark this answers is not in the document]
Comment (fingolfin) [reply src/flint/fq_default.jl:850]: @lgoettgens just decide if you'd like to apply this suggestion or not, then merge :-)

**Pull request merged**